taipy/core/data/_file_datanode_mixin.py

```diff
@@ -212,3 +220,23 @@ def _migrate_path(self, storage_type, old_path) -> str:
         if os.path.exists(old_path):
             shutil.move(old_path, new_path)
         return new_path
+
+    def _duplicate_file(self, id: str) -> Optional[str]:
+        if os.path.exists(self.path):
+            folder_path, base_name = os.path.split(self.path)
+
+            if base_name.startswith(self.__TAIPY_DUPLICATED_PREFIX):
+                base_name = "".join(base_name.split("_")[5:])
+            new_base_path = os.path.join(folder_path, f"{self.__TAIPY_DUPLICATED_PREFIX}_{id}_{base_name}")
+
+            if os.path.isdir(self.path):
+                shutil.copytree(self.path, new_base_path)
+            else:
+                shutil.copy(self.path, new_base_path)
+
+            if hasattr(self._properties, "_entity_owner"):  # type: ignore[attr-defined]
+                del self._properties._entity_owner  # type: ignore[attr-defined]
+            self._properties[self._PATH_KEY] = new_base_path  # type: ignore[attr-defined]
+
```

Comment:
I think we can add it back right after setting all the values to the `self._properties` here, the new `_entity_owner` should be the new data node id I suppose